teuthology/test/test_repo_utils.py

```diff
@@ -12,20 +13,23 @@
 
 
 class TestRepoUtils(object):
-    src_path = '/tmp/empty_src'
-    # online_repo_url = 'https://github.com/ceph/teuthology.git'
-    # online_repo_url = 'git://ceph.newdream.net/git/teuthology.git'
-    online_repo_url = 'https://github.com/ceph/empty.git'
-    offline_repo_url = 'file://' + src_path
-    repo_url = None
-    dest_path = '/tmp/empty_dest'
 
     @classmethod
-    def setup_class(cls):
+    def setup_class(self):
```

Comment:
why s/cls/self/ ? i feel `self` in this context is a kinda misleading.

Reply:
> why s/cls/self/ ? i feel `self` in this context is a kinda misleading.
There is no difference which is name of the first argument from one side, from other side since all test_ methods have 'self', it is better to be consistent and use the same naming.

Reply:
please note the `self` here is not an instance of `TestRepoUtils`. and `setup_class()` is not a test method.

Reply:
> please note the `self` here is not an instance of `TestRepoUtils`. and `setup_class()` is not a test method.
What is it if not the test class instance?

Reply:
@tchaikov I addressed you request in the recent push.

Reply:
probably you could take a look at https://docs.python.org/3/library/functions.html#classmethod ? and see also, https://github.com/pytest-dev/pytest/blob/5ca81596bbbb35da0b8b742afaf7ed2ef075ccda/src/_pytest/python.py#L732 , i don't have enough bandwidth looking into the details. but the gist is that it is very likely to be an instance of the class. and we cannot assume that it is an instance of `TestRepoUtils`.

Reply:
> @tchaikov I addressed you request in the recent push.

@tchaikov ^